src/bindings/python/flux/job/list.py

```diff
@@ -178,5 +249,15 @@ def fetch_jobs(self):
         )
 
     def jobs(self):
-        for job in self.fetch_jobs().get_jobs():
-            yield JobInfo(job)
+        """Synchronously fetch a list of JobInfo objects from JobList query
+
+        If the Future object returned by JobList.fetch_jobs has not yet been
+        fulfilled (e.g. is_ready() returns False), then this call may block.
+        Otherwise, returns a list of JobInfo objects for all jobs returned
+        from the underlying job listing RPC.
+        """
+        rpc = self.fetch_jobs()
```

Comment:
should the variable `rpc` be renamed to something else?  b/c its an RPC or future?  not sure what it would be.